lib/owncloud/test_sharePropagation.py

```diff
@@ -0,0 +1,256 @@
++-------------+-------------------------+-------------------------+----------------------+
+| 5           |                         | upload in shared dir    |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 6           | verify propagation      | verify propagation      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 7           | unshare folder          |                         |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 8           | verify etag is the same | verify propagation      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 9           | share folder with R2 R3 |                         |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 10          |                         | R2 reshare with R4      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 11          | verify etag is the same | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 12          |                         | R2 upload in shared dir |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 13          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 14          |                         |                         | upload in shared dir |
++-------------+-------------------------+-------------------------+----------------------+
+| 15          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 16          |                         | R2 unshares folder      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 17          | verify etag is the same | verify etag is the same | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+"""
+
+from smashbox.utilities import *
+import itertools
+import os.path
+import re
+
+def parse_worker_number(worker_name):
+    match = re.search(r'(\d+)$', worker_name)
+    if match is not None:
+        return int(match.group())
+    else:
+        return None
+
+@add_worker
+def setup(step):
+
+    step(1, 'create test users')
+    reset_owncloud_account(num_test_users=4)
+    check_users(4)
+
+    reset_rundir()
+    reset_server_log_file()
```

Comment:
```py
    reset_rundir()
    reset_server_log_file()
```
can be removed, smashbox is automatically doing that before each test since end of september